### PR TITLE
remove evals

### LINKED ISSUE #105: replace evaled functions
See discussion here: https://github.com/JuliaGeo/LibGEOS.jl/pull/95#issuecomment-1058545908

### PR DESCRIPTION
**Pull request by visr**
Fixes #105

This replaces this pattern:

```julia
for geom in (
    :Point,
    :MultiPoint,
    :LineString,
    :MultiLineString,
    :LinearRing,
    :Polygon,
    :MultiPolygon,
    :GeometryCollection,
)
    @eval getXMin(obj::$geom, context::GEOSContext = _context) = getXMin(obj.ptr, context)
end
```

With this:
```julia
getXMin(obj::Geometry, context::GEOSContext = _context) = getXMin(obj.ptr, context)
```

Where `Geometry` was already defined as:
```julia
const Geometry = Union{
    Point,
    MultiPoint,
    LineString,
    MultiLineString,
    LinearRing,
    Polygon,
    MultiPolygon,
    GeometryCollection,
}
```

So instead of having one method per geometry type, there is just one for all of them. According to https://github.com/JuliaGeo/LibGEOS.jl/pull/95#issuecomment-1058868759 it specializes fine.

Everything except the last one is a simple replace that shouldn't change any behavior. The last one is separated in 3c9a7f7fe562d402f168c9a95da688afc063da53 and should allow some new functionality.

### CODE REVIEW

File: src/geos_functions.jl [hunk 1]
```diff
@@ -1352,7 +1349,7 @@ The precision reduced result. Caller must free with `GEOSGeom_destroy()` NULL on
 function setPrecision(
     geom::GEOSGeom,
     gridSize::Real,
-    flags::GEOSPrecisionRules,
+    flags::Union{GEOSPrecisionRules,Integer},
```
Comment (evetion) [inline src/geos_functions.jl:1352]: Why this change without the check?

**Review by evetion — APPROVED** [review]

File: src/geos_functions.jl (hunk 1 above)
Comment (visr) [reply src/geos_functions.jl:1352]: Because the check disallows bitwise or combined flags, see also the last commit message.

Comment (evetion) [reply src/geos_functions.jl:1352]: Wouldn't those bitflags be an `UInt8` instead of `Integer`?

Comment (visr) [reply src/geos_functions.jl:1352]: `ccall` will convert any `Integer` to the `Cint` that it ultimately needs. I added tests for both `UInt8` and the enum bitflags, which become `UInt32`. And `Int` was already tested.
